### PR TITLE
9 intensity normalisation transform

### PR DESCRIPTION
**Pull request by Nic-Ma**
Hi All,

This PR implemented the intensity normalization transform for task #9 
Designed according to our latest discussion:
_1. input data is dict format with keys for fields.
2. only based on PyTorch and data shape is channel_last._

Could you please help review it when you are available?
Thanks in advance.

I verified this PR locally with the below program:
```py
# test 1 key with auto subtrahend and divisor
normalizer = IntensityNormalizer(apply_keys=('image',))
img = np.arange(27).reshape(3, 3, 3).astype(np.float32)
label = np.arange(27).reshape(3, 3, 3).astype(np.float32)
data = {'image': img, 'label': label}
print(data)
print(normalizer(data=data))

# test 2 keys with specified subtrahend and divisor
normalizer = IntensityNormalizer(
    apply_keys=('image', 'label'),
    subtrahend=np.ones((3), dtype=np.float32),
    divisor=np.ones((3), dtype=np.float32)
)
img = np.arange(27).reshape(3, 3, 3).astype(np.float32)
label = np.arange(27).reshape(3, 3, 3).astype(np.float32)
data = {'image': img, 'label': label}
print(data)
print(normalizer(data=data))
```

### CODE REVIEW

**Comment by wyli** [comment]
please update based on https://github.com/Project-MONAI/MONAI/issues/9#issuecomment-576265046

**Comment by Nic-Ma** [comment]
Hi @wyli @ericspod @atbenmurray ,

I already update the code and PR description according to our latest discussion.
Could you please help review it again?
Thanks in advance.

**Comment by wyli** [comment]
@ericspod could you review this PR? mainly added a minimal transform interface, and updated `ImageTestCase` to have `NumpyImageTestCase2D` and `TorchImageTestCase2D `.

**Comment by Nic-Ma** [comment]
Hi @ericspod @atbenmurray @wyli ,

I updated this PR to remove dict-key processing and be compatible with Eric's other transforms.
Could you please help review again?
Thanks in advance.

File: tests/test_intensity_normalizer.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+class IntensityNormTestCase(NumpyImageTestCase2D):
+
+    def test_image_normalizer_default(self):
+        normalizer = IntensityNormalizer()
+        normalised = normalizer(self.imt)
+        expected = (self.imt - np.mean(self.imt)) / np.std(self.imt)
+        self.assertTrue(np.allclose(normalised, expected))
+
+
+if __name__ == '__main__':
```
Comment (ericspod) [inline tests/test_intensity_normalizer.py]: Generally you don't need a main section in the test case scripts, you can run tests from the root directory with 

    python -m unittest test/test_intensity_normalizer.py

It doesn't hurt to be here though.

**Comment by ericspod** [comment]
Beyond the issue with the CI pipeline not completing the code is good to go.

**Comment by Nic-Ma** [comment]
> Beyond the issue with the CI pipeline not completing the code is good to go.

Hi @ericspod ,

Thanks for your quick review!
I think maybe gitlab-staging server is unavailable now.
The gitlab-master server passed tests.
Of course, I am OK if you think it's better to wait for the stage server recovery.
Thanks.

**Comment by ericspod** [comment]
I would just go ahead with the merge.

**Comment by Nic-Ma** [comment]
> I would just go ahead with the merge.

Cool, thanks!
Could you please help merge it directly?

File: tests/test_intensity_normalizer.py (hunk 1 above)
Comment (wyli) [reply tests/test_intensity_normalizer.py]: Good point! @ben could you please update the contribution guidelines about running all unit tests and single unit test . PR for new features should include new unit tests and inherit test case base classes.

**Pull request merged**